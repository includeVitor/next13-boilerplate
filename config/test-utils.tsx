// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import { render, RenderOptions } from "@testing-library/react";
import { FC, ReactElement } from "react";

type ProvidersProps = {
    children: React.ReactNode;
};

const Providers: FC<ProvidersProps> = ({ children }) => {
    return <>{children}</>;
};

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, "queries">
) => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";
// override render method
export { customRender as render };
