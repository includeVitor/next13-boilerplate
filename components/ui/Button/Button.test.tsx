import { render } from '@config/test-utils'
import { Button } from '@components/ui/index'

describe('Button', () => {
    it('Should be able to match snapshot', () => {
        const { container } = render(<Button>Simple Button</Button>)
        expect(container.firstChild).toMatchSnapshot()
    })
})
