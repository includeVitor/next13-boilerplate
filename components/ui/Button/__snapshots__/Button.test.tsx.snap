// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should be able to match snapshot 1`] = `
<button>
  Simple
</button>
`;
